client/app/account/hooks/useChangePassword.ts

```diff
@@ -0,0 +1,56 @@
+import { useRouter } from 'next/navigation'
+import { useQueryClient } from '@tanstack/react-query'
+import { useForm } from 'react-hook-form'
+import { useToast } from '@/_hooks'
+import { isErrorWithMessage } from '@/_utils'
+import { useChangePasswordMutation } from './api'
+import { PasswordSchema, passwordResolver } from '../schema'
+
+export const useChangePassword = () => {
+  const queryClient = useQueryClient()
+  const router = useRouter()
+  const { toast } = useToast()
+
+  const form = useForm<PasswordSchema>({
+    resolver: passwordResolver,
+    defaultValues: {
+      currentPassword: '',
+      newPassword: '',
+      confirmNewPassword: '',
+    },
+  })
+
+  const { mutate, isPending } = useChangePasswordMutation()
+
+  const onUpdateAccount = (data: PasswordSchema) => {
+    mutate(data, {
+      onSuccess: () => {
+        queryClient.invalidateQueries({ queryKey: ['me'] })
```

Comment:
It's better to say `onChansePassword` for the function name. And you can delete the `queryClient.invalidateQueries({ queryKey: ['me'] })` as we did not store in the cache!

Reply:
Fixed it.
8d19e953ba556e4fe9875e45cc47c29408c60f3d